### ISSUE TITLE
Reduce stack usage of unit tests

### ISSUE THREAD

**Issue by tcarmelveilleux** [project-chip/connectedhomeip#19951]
#### Problem

The following pattern is omnipresent in unit tests:

```
int TestCommandInteractionTest()
{
    TestContext gContext;
    nlTestRunner(&sSuite, &gContext);
    return (nlTestRunnerStats(&sSuite));
}
```

In this boilerplate, the `TestContext` can be very large, especially if it accounts for networking and other protocol stack resources. When individual tests are complex, they can have code paths that appear to take more than the max configured stack (see `-Wstack-usage=8192` in `build/config/compiler/BUILD.gn`).

These stack measurements are not actually done on the platform using stack in the case of tests, but rather from a x64 compiler, which increases the stack usage artificially.

 #### Proposed Solution
- Move contexts to heap via unique_ptr to reduce stack usage
- Find more accurage ways of setting the stack limit

**Comment by tcarmelveilleux** [comment]
See #19931

**Comment by tcarmelveilleux** [comment]
This is resolved enough since #19931
